File: oskelly-ios/oskelly/src/swift/data/Attribute.swift
//
// Created by Виталий Хлудеев on 18.10.17.
// Copyright (c) 2017 Виталий Хлудеев. All rights reserved.
//

import Foundation

class Attribute {

    let id: Int
    let name: String
    var values: [AttributeValue]
    var selectedValue: AttributeValue?

    init(
            id: Int,
            name: String,
            values: [AttributeValue],
            selectedValue: AttributeValue?
    ) {
        self.id = id
        self.name = name
        self.values = values
        self.selectedValue = selectedValue
    }
}

File: oskelly/src/main/java/su/reddot/domain/model/attribute/Attribute.java
package su.reddot.domain.model.attribute;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Getter @Setter
public class Attribute {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @OneToMany(mappedBy = "attribute", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<AttributeValue> attributeValues;
}
